### ISSUE TITLE
Cydia does not appear after restarting

### ISSUE THREAD

**Issue by dragon05jailbreak** [kpwn/yalu102#341]
I have an iphone 6 with version 10.2 when I open yalu once downloaded and press GO after the restart appears Cydia or tried 3 times but nothing help me!

**Comment by DarknesGaming** [comment]
#162 
This has been referenced for 2 reasons:
1. This has nothing to do with the code itself, nor is this a code-related issue. Everything mostly works fine, and I have an iPhone 6 with iOS 10.2 that works fine with Yalu. You might be on iOS 10.2.1.
2. Chances are you're using beta 1 or 2 of the Yalu jailbreak. Please use beta 3 or above if you want this jailbreak to work.
Also, if the jailbreak succeeds, it does not "restart." A restart is where you see that Apple logo. The jailbreak simply "resprings your device", which is where you don't see the Apple logo. You only see the loading animation instead.

**Comment by dragon05jailbreak** [comment]
I tried with the beta 7, but after the restart nothing

**Comment by cdecker15** [comment]
@dragon05jailbreak you're giving very little details on what exactly the issue is. Can you double check that you are running iOS 10.2 (not 10.2.1), also ensure that you have completely removed the any previous version of Yalu prior to sideloading the Beta 7 IPA. Need more details rather than just a "help me".

**Comment by oiklo** [comment]
@dragon05jailbreak hello!
I have the same problem as you, after try everything i have check my version of ios, i am running ios 10.2, but i still have a beta version.
So i think it doesn't work with ios beta :(

**Comment by jtv7** [comment]
@oiklo You probably have to update the offsets in offsets.c, then it might work

**Comment by dragon05jailbreak** [comment]
Can you give me more details I downloaded the first beta version of Yalu I have not made a reset on the phone, how do I remove the beta previous?

**Comment by jtv7** [comment]
@dragon05jailbreak Just delete it and re-install the latest beta of yalu.

**Comment by dragon05jailbreak** [comment]
I had already done this I had deleted from home

**Comment by dragon05jailbreak** [comment]
What should I Do Not Restore Can I do this anymore because Apple has stopped signing 10.2

**Comment by oiklo** [comment]
@jtv7
What should i add in offsets.c for ios 10.2 beta 3 (build 14c5077b)
Thanks

**Comment by jtv7** [comment]
@oiklo If you decrypt the kernel and send it to me, I can give you the offsets

**Comment by dragon05jailbreak** [comment]
@jtv7 What do I owe you? Then?

**Comment by jtv7** [comment]
@dragon05jailbreak What do you mean?

**Comment by master3395** [comment]
I have the same issue, iphone 6, ios 10.2
The phone auto restarts after hit jailbreak on the app.
And after the first time you start cydia, if you had to restart the phone, your stuck with cydia crashing every time.

**Comment by dragon05jailbreak** [comment]
@jtv7 that is, I want to know what I have to do to solve this problem e'poter download the jailbreak

**Comment by jtv7** [comment]
@dragon05jailbreak Replace offsets.c in the project with this file.  Then recompile
https://drive.google.com/file/d/0Bw-eKoFAPzPqalRjMW56WkZ0OGM/view?usp=sharing

**Comment by master3395** [comment]
@jtv7 Do you know a place were it's already compiled? as i can't download the apple program to compile it.

**Comment by oiklo** [comment]
@jtv7
Will this offsets work for me?
I d'ont know how to decrypt the kernel :(

**Comment by dragon05jailbreak** [comment]
@jtv7 
I have to do it with xcode?

**Comment by tangalbert919** [comment]
@dragon05jailbreak, please specify the build number of your iOS version. A build number looks like the one on the parentheses (14C92), and I just referenced iOS 10.2 right there.
Try re-installing the Yalu 7 beta thing again, and if it STILL doesn't work, keep trying until you've tried 3 times.
@jtv7 those are the same offsets we can find in this repository.
Send me the kernel, and I can get the offsets. I'll make sure this ends up in my fork, and send a pull request to @kpwn (if he actually gives a s*** about beta versions of iOS, and chances are he doesn't).

**Comment by jtv7** [comment]
@oiklo Here is a compiled ipa with the offsets of the beta firmware you gave me earlier: https://drive.google.com/file/d/0Bw-eKoFAPzPqVURrMENBX2ZTU1U/view?usp=sharing

You have to install with Cydia Impactor.  Let me know if this works and i will do a pull request with the offsets for the particular beta you are on (if he accepts it).  Also, don't share this anywhere (im pulling the link as soon as you tell me if it works) because its frowned upon to install this ipa from an unknown source (me).

**Comment by jtv7** [comment]
@tangalbert919 What do you mean the same offsets?

**Comment by jtv7** [comment]
@oiklo I didn't see your previous comment.  Yes, the offsets.c file I included is for you, if you can compile it yourself.

**Comment by tangalbert919** [comment]
Never mind. I just went to the bottom of that document from your Google drive. I see two new offsets, but I don't know what version it's for (I'm going to assume it's iOS 10.2.1).

**Comment by jtv7** [comment]
@tangalbert919  iPhone 6 iOS 10.2 beta 3 (build 14c5077b)

**Comment by tangalbert919** [comment]
I have iOS 10.2 beta 2 (build 14c5069c) for that same device.
@dragon05jailbreak still waiting on that build number.

**Comment by jtv7** [comment]
@tangalbert919 Do you want the offset for that device (build 14c5069c)?

**Comment by jtv7** [comment]
@tangalbert919 Here is offsets.c file with both betas included:
https://drive.google.com/file/d/0Bw-eKoFAPzPqalRjMW56WkZ0OGM/view?usp=sharing

**Comment by DarknesGaming** [comment]
Wow. You're faster than I am. That's mostly because I still can't find the kernel base, so I have to use a program that can find the allproc offset.

**Comment by oiklo** [comment]
@jtv7
I've done it with your compiled ipa with offsets fort ios 10.2 beta 3,
It's perfectly working, thanks a lot, cydia is launching and working !

**Comment by GR3YH4TT3R** [comment]
Why are you guys trying to jailbreak and use a beta iOS version??? You should NEVER stay on an iOS beta for your daily jailbroken iDevices they're BETAS for a reason

**Comment by GR3YH4TT3R** [comment]
Beta OS + beta jailbreak = problems guaranteed

**Comment by DarknesGaming** [comment]
Problems are always guaranteed, @GR3YH4TT3R. Nobody said this would work out-of-the-box.
Plus, they don't have a choice. If they want to use the jailbreak, they have to stay on a beta iOS version.

**Comment by GR3YH4TT3R** [comment]
How so? I've used betas with no problems at all but when you run beta software on a beta OS you're guaranteed to have problems. And I'm not on a beta iOS version and 99.99% of those jailbreaking aren't on a beta iOS (exactly why there aren't any beta iOS offsets in the master branch) beta iOS builds are buggy and inherently unreliable themselves and only intended for developers to run on test devices or in an emulator. Anyone who's on a beta iOS version should've known to update to the master release (ie iOS 10, 10.0.1, 10.1, 10.1.1, 10.2) not iOS 10.2 beta 4 remember betas come before the master release and iOS 10.2 is capable of being jailbroken (unless you're on an i7/i7+) so there was absolutely no reason to stay on a 10.2 beta and not update to 10.2 and if they weren't smart enough to understand that and update to a stable jailbreakable iOS release then that's their own fault and any instabilities or incompatibilities because of staying on a beta release is their own stupidity.

**Comment by LeGitz** [comment]
I am on 10.2 (iPad air 1) i restored it (reset delete all - settings -Apps-...) i installed beta 7 pressed go but nothing Happens After reboot

**Comment by LeGitz** [comment]
Please help

**Comment by OGKevin** [comment]
U should not have done that. I think ur doomed. There were tons of warnings to not use these methods.

**Comment by KILLCAMPER** [comment]
I have no problem with mine if fails 1st time hold power + home and the go again if you see storage full for split sec it in jb state it works for me hope it will for you have tested it and re jb about 5 times after this method there is a low % on iPad Air 2 jb success

**Comment by LeGitz** [comment]
what warning what methode ? do ineed to reload my backup?

**Comment by LeGitz** [comment]
@OGKevin pls help me idk what do/how to fix

**Comment by karandeepdps** [comment]
can you provide ipa for  iOS 10.2 beta 4

**Comment by kpwn** [comment]
this post is a great example of how to get blocked from this repo.

**Comment by jtv7** [comment]
@kpwn Sorry, bad judgement on my part 👎
